### ISSUE TITLE
Alerts for code host rate limiting, time spent waiting on rate limiting, etc.

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#8100]
For GitHub, we have a Prometheus metric that tells us information about rate limiting. We do not have this in place for ANY other code host. For each code host, there should be metrics in place for:

1. Consumed rate limit
2. How much time we spent waiting for rate limiting in total
3. How much time we spent waiting for rate limiting per-request
4. How many requests we sent to the code host
5. How long requests to the code host took

It is frankly surprising that we do not have these metrics in place.

**Comment by slimsag** [comment]
See for example https://github.com/sourcegraph/sourcegraph/issues/7262 which indicates this was happening on Sourcegraph.com for a long period of time completely unnoticed.

**Comment by slimsag** [comment]
Some additional details in https://github.com/sourcegraph/sourcegraph/issues/8581

**Comment by pecigonzalo** [comment]
I believe this should be assigned to Cloud /cc @tsenart

**Comment by tsenart** [comment]
@ryanslade: Assigning to you for triage because I think you have the most context on what here already exists, what should still be done and what we can probably punt on.

**Comment by ryanslade** [comment]
We recently started tracking tokens remaining in `repo-updater` and we do so for `GitHub` and `GitLab` which are the only code hosts that return rate limit information in their headers. See the `External Services` sections [here](https://sourcegraph.com/-/debug/grafana/d/repo-updater/repo-updater?orgId=1). We are tracking the limit per external service here and on `sourcegraph.com` we are only tracking our "global" external services.. the ones we use by default for all public repos.

We do also have internal rate limiting in place for `GitHub`, `GitLab` and `BitbucketServer/Cloud`. The intention of these rate limiters is to allow admins to set an upper limit on the rate of requests that Sourcegraph will make to each code host and are not tied to any specific user or token. The reason we haven't done any monitoring on these is that they are not used much AFAIK and also they are not global since they are in memory in repo-updater. This means that they are reset every time repo-updater is restarted. I added these about a year ago and I would actually be interested to know if they're being used by anyone since I feel they don't actually add that much value and confuse things by having two layers of rate limiting. 

As a first step I think we should instrument the metrics mentioned above for `GitHub` and `GitLab` via the current mechanism of hooking into our rate limit monitor.

What would be the best way to discover if anyone is using our internal rate limiters? They are configured via external service config. We could perhaps use pings but that feels a little heavy.. How have we done this in the past?
